File: Source/code/frames/frameFontPanel.pas
// ===================================================================
//
// (c) Paul Alan Freshney 2012-2022
// www.freshney.org :: paul@freshney.org :: maximumoctopus.com
//
// https://github.com/MaximumOctopus/LEDMatrixStudio
//
// Please do not modifiy this comment section

//
// ===================================================================

unit frameFontPanel;


interface


uses
  Winapi.Windows, Winapi.Messages, System.SysUtils, System.Variants, System.Classes,
  Vcl.Graphics, Vcl.Controls, Vcl.Forms, Vcl.Dialogs, Vcl.StdCtrls, Vcl.Buttons;


type
  TframeFont = class(TFrame)
    sbChangeFont: TSpeedButton;
    lFont: TLabel;
    fdMain: TFontDialog;
    procedure lFontClick(Sender: TObject);
  private
    FFontAscii : integer;
  public
    procedure SetFont(aNewAscii : integer);
  end;


implementation


{$R *.dfm}


procedure TframeFont.lFontClick(Sender: TObject);
begin
  if fdMain.Execute then begin
    lFont.Font := fdMain.Font;

    sbChangeFont.Caption := fdMain.Font.Name;
  end;
end;


procedure TframeFont.SetFont(aNewAscii : integer);
begin
  FFontAscii    := aNewAscii;

  lFont.Caption := Char(aNewAscii);
end;


end.

File: Source/Frames/FrameFontPanel.cpp
// ===================================================================
//
//   (c) Paul Alan Freshney 2012-2024
//   www.freshney.org :: paul@freshney.org :: maximumoctopus.com
//
//   https://github.com/MaximumOctopus/LEDMatrixStudio
//
//   https://maximumoctopus.hashnode.dev/
//
//   C++ Rewrite October 11th 2023
//
// ===================================================================

#include <vcl.h>
#pragma hdrstop

#include "FrameFontPanel.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TframeFont *frameFont;

__fastcall TframeFont::TframeFont(TComponent* Owner)
	: TFrame(Owner)
{
}


void __fastcall TframeFont::lFontClick(TObject *Sender)
{
	if (fdMain->Execute())
	{
		lFont->Font = fdMain->Font;

		sbChangeFont->Caption = fdMain->Font->Name;
	}
}


void TframeFont::SetFontAscii(int ascii)
{
	FontAscii = ascii;

	lFont->Caption = Char(ascii);
}
